### PR TITLE
ref(reader): Move reader instantiation to `snuba.environment`

### PR DESCRIPTION
**Pull request by tkaemming**
This removes the `reader` parameter from `raw_query`, since there is no way to actually set it without modifications to `parse_and_run_query` and correspondingly `split_query` and it's inner functions.

The HTTP reader will have state (the HTTP connection pool), so reinstantiating it on each call to `parse_and_run_query` isn't ideal — it will be better to instantiate a `Reader` instance and reuse it. Since the ClickHouse cluster address is a global setting, it (unfortunately) seems appropriate to make this global state, similar to the native driver pool.

I think that in the future, we'll want the table storages themselves to provide the writer and reader interfaces (since I assume we'll want table storages to be able to reference different ClickHouse clusters) but that seems like a reasonably far future.

### CODE REVIEW

File: snuba/environment.py [hunk 1]
```diff
@@ -35,3 +37,5 @@ def setup_sentry() -> None:
     settings.CLICKHOUSE_PORT,
     client_settings={"readonly": True},
 )
+
+reader: Reader[ClickhouseQuery] = NativeDriverReader(clickhouse_ro)
```
Comment (fpacifici) [inline snuba/environment.py]: This will be reused only on split queries or any other request that runs more than one query. Is this correct ?

**Review by fpacifici — APPROVED** [review]

File: snuba/environment.py (hunk 1 above)
Comment (tkaemming) [reply snuba/environment.py]: No, this is a global and instantiated at startup

Comment (fpacifici) [reply snuba/environment.py]: So I guess we will have to ensure thread safety as well when we get to the http reader if that is stateful right ?

Comment (tkaemming) [reply snuba/environment.py]: Yes

**Pull request merged**